### PR TITLE
Preview Dialing Changes

### PR DESCRIPTION
**Pull request by vaibhavkhanna08**

### CODE REVIEW

**Review by devikasuresh20 — APPROVED** [review]

File: src/app/app-modules/associate-anm-mo/outbound-worklist/outbound-worklist.component.ts [hunk 1]
```diff
@@ -170,6 +177,20 @@ export class OutboundWorklistComponent implements OnInit {
           ];
           this.dataSource.paginator = this.paginator;
           this.dataSource.sort=this.sort;
+
+          if(this.isAutoPreviewDial === true && this.associateAnmMoService.isStartAutoPreviewDial === true && this.dataSource.data.length > 0 && this.associateAnmMoService.autoDialing === false) {
```
Comment (drtechie) [inline src/app/app-modules/associate-anm-mo/outbound-worklist/outbound-worklist.component.ts:181]: @vaibhavkhanna08 @devikasuresh20 
Why not use

if(this.isAutoPreviewDial && this.associateAnmMoService.isStartAutoPreviewDial  && this.dataSource.data.length > 0 && !this.associateAnmMoService.autoDialing)

File: src/app/app-modules/associate-anm-mo/outbound-worklist/outbound-worklist.component.ts [hunk 2]
```diff
@@ -208,6 +229,17 @@ export class OutboundWorklistComponent implements OnInit {
           ];
           this.dataSource.paginator = this.paginator;
           this.dataSource.sort=this.sort;
+          if(this.isAutoPreviewDial === true && this.associateAnmMoService.isStartAutoPreviewDial === true && this.dataSource.data.length > 0 && this.associateAnmMoService.autoDialing === false) {
```
Comment (drtechie) [inline src/app/app-modules/associate-anm-mo/outbound-worklist/outbound-worklist.component.ts:232]: Same comment as above.

File: src/app/app-modules/associate-anm-mo/outbound-worklist/outbound-worklist.component.ts [hunk 3]
```diff
@@ -170,6 +177,20 @@ export class OutboundWorklistComponent implements OnInit {
...
           this.dataSource.sort=this.sort;
+
+          if(this.isAutoPreviewDial === true && this.associateAnmMoService.isStartAutoPreviewDial === true && this.dataSource.data.length > 0 && this.associateAnmMoService.autoDialing === false) {
+            
+            this.isChecked = true;
+            let previewTime = this.previewWindowTime * 1000;
+            
+          this.autoPreviewTimeSub = timer(previewTime).pipe( 
+            map(() => {   
+            this.StartAutoPreviewDialing(true);
```
Comment (drtechie) [inline src/app/app-modules/associate-anm-mo/outbound-worklist/outbound-worklist.component.ts:188]: Indent is off.

File: src/app/app-modules/associate-anm-mo/agents-innerpage/agents-innerpage.component.ts [hunk 4]
```diff
@@ -66,6 +66,7 @@ export class AgentsInnerpageComponent implements OnInit {
   isEnableComp: any = "Outbound Worklist";
   callTypes:any[]=[];
   callTypeId:any;
+  previousAgentState:any;
```
Comment (drtechie) [inline src/app/app-modules/associate-anm-mo/agents-innerpage/agents-innerpage.component.ts:69]: Why is this any?
This is a general comment.
Lost all the benefits of TypeScript

**Review by drtechie — COMMENTED** [review]
Can be improved.
@vaibhavkhanna08 @devikasuresh20

**Pull request merged**